src/web/components/App.tsx

```diff
@@ -399,6 +399,38 @@ export const App = ({ CAPI, NAV }: Props) => {
         CAPI.config.isSensitive,
     ]);
 
+    // ************************
+    // *   Google Analytics   *
+    // ************************
+    useEffect(() => {
+        // TODO: handle when consent removed
+        onConsentChange((state: any) => {
+            const consentGiven = getConsentFor('google-analytics', state);
+            if (consentGiven) {
+                const gaScript = document.createElement('script');
+                gaScript.id = 'ga-script';
+                gaScript.type = 'text/javascript';
+                gaScript.src = 'https://www.google-analytics.com/analytics.js';
+                document.head.appendChild(gaScript);
+
+                const gaCustomScript = document.createElement('script');
+                gaCustomScript.id = 'ga-custom-script';
+                gaCustomScript.type = 'text/javascript';
+                gaCustomScript.src = window.guardian.gaPath;
+                document.head.appendChild(gaCustomScript);
+
+                console.log(`gaConsent state: ${consentGiven}`);
+            } else {
+                const gaScriptElement = document.getElementById('ga-script');
+                gaScriptElement?.parentNode?.removeChild(gaScriptElement);
+                const gaCustomScriptElement = document.getElementById('ga-custom-script');
+                gaCustomScriptElement?.parentNode?.removeChild(gaCustomScriptElement);
+
```

Comment:
This doesn't really do anything, because by the time you're running this script the contents of  is loaded into the JS engine and it is divorced from the script tag on the page. You probably want to no-op window.ga like in frontend?

Reply:
I think I had the same doubts as you but I seem to remember when we were testing that this code proved itslef out, that no further requests were sent. But maybe we didn't go back and test the negative...

Reply:
Here's an example of what I think you're doing - run some JS (i.e ping it on the window) and then remove the script - I don't think the script removal does anything once the JS is loaded  https://codesandbox.io/s/hungry-montalcini-z5ped?file=/src/index.js

Reply:
Thanks both, we took another look at this and sure enough we weren't removing the scripts, but the script removal was redundant since on a page reload we check the consent status and either add the scripts or not depending on consent. We've removed the attempt to remove the scripts now.